torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py

```diff
@@ -57,11 +66,36 @@ def fp16_compress_hook(
         compressed_tensor, group=group_to_use, async_op=True
     ).get_future()
 
-    def decompress(fut):
-        decompressed_tensor = bucket.get_tensors()[0]
-        # Decompress in place to reduce the peak memory.
-        # See: https://github.com/pytorch/pytorch/issues/45968
-        decompressed_tensor.copy_(fut.value()[0].div_(world_size))
-        return [decompressed_tensor]
+    decompress_fp16 = decompress_fp16_callable(bucket)
+    return fut.then(decompress_fp16)
+
+def fp16_compress_wrapper(hook: callable) -> callable:
+    """
+    This wrapper applies ``float 32`` to ``float 16`` compression to an arbitrary DDP communication hook.
+
+    Example::
+    >>> state = PowerSGDState(process_group=process_group, matrix_approximation_rank=1, start_powerSGD_iter=10)
+    >>> ddp_model.register_comm_hook(state, fp16_compress_wrapper(powerSGD_hook))
+    """
+
+    def comm_hook_with_fp16(hook_state, bucket: dist.GradBucket):
+        # Cast tensors to FP16. TODO: Optimize this as per https://github.com/pytorch/pytorch/pull/46078/files.
+        bucket_tensors = bucket.get_tensors()
+        bucket_tensors[0] = bucket_tensors[0].to(torch.float16)
+
+        fut = hook(hook_state, bucket)
+
+        # Some hooks are given the ProcessGroup instance directly, whereas others (e.g. powerSGD_hook) are given a state object.
```

Comment:
I think you could consider a refactor by creating a base State class with `processor_group` as an instance variable. The powerSGD hook could then inherit from the base class. It would clean up this code and make it so that all of the hooks accept the same class type as the first arg. 

Although, I thought that would be overkill for this PR.